include/ddc/for_each.hpp

```diff
@@ -120,6 +120,12 @@ inline void for_each(DiscreteDomain<DDims...> const& domain, Functor&& f) noexce
     for_each(default_policy(), domain, std::forward<Functor>(f));
 }
 
+template <class... DDims, class Functor>
+inline void for_each(DiscreteVector<DDims...> const& domain, Functor&& f) noexcept
```

Comment:
Why do we need this `for_each` version ?

Reply:
I am skeptical, is it for the periodic boundary conditions ? Could we imagine a remapping instead ?

Reply:
Maybe call it differently then ? I think the matching of the domains can help debugging.

Reply:
We could, but I figured that the use of a different type (`DiscreteVector<>` instead of `DiscreteDomain<>`) was enough